Makefile

```diff
@@ -1,85 +1,100 @@
+          --isfinal --buildarch=$(ARCH) --volid=$(IMAGE_NAME)-$(ARCH)-$(VERSION) \
+          $(_LORAX_ARGS) \
           --repo /etc/yum.repos.d/fedora.repo \
           --repo /etc/yum.repos.d/fedora-updates.repo \
-          --add-template $(base_dir)/lorax_templates/set_installer.tmpl \
-		  --add-template $(base_dir)/lorax_templates/configure_upgrades.tmpl \
-          $(base_dir)/results/
-	mv $(base_dir)/results/images/boot.iso $(base_dir)/
+          --add-template $(_BASE_DIR)/lorax_templates/set_installer.tmpl \
+		  --add-template $(_BASE_DIR)/lorax_templates/configure_upgrades.tmpl \
+          $(_BASE_DIR)/results/
+	mv $(_BASE_DIR)/results/images/boot.iso $(_BASE_DIR)/
 
-container/$(image_name)-$(image_tag):
+# Step 3: Download container image
+container/$(IMAGE_NAME)-$(IMAGE_TAG):
 	mkdir container
-	podman pull $(image_repo)/$(image_name):$(image_tag)
-	podman save --format oci-dir -o $(base_dir)/container/$(image_name)-$(image_tag) $(image_repo)/$(image_name):$(image_tag)
-	podman rmi $(image_repo)/$(image_name):$(image_tag)
+	podman pull $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
+	podman save --format oci-dir -o $(_BASE_DIR)/container/$(IMAGE_NAME)-$(IMAGE_TAG) $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
+	podman rmi $(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
 
 install-deps:
 	dnf install -y lorax xorriso podman git rpm-ostree
 
 
-
+# Step 1: Generate Lorax Templates
 lorax_templates/%.tmpl: lorax_templates/%.tmpl.in
-	sed 's/@IMAGE_NAME@/$(image_name)/'                        $(base_dir)/lorax_templates/$*.tmpl.in > $(base_dir)/lorax_templates/$*.tmpl
-	sed 's/@IMAGE_REPO@/$(image_repo_escaped)/'                $(base_dir)/lorax_templates/$*.tmpl > $(base_dir)/lorax_templates/$*.tmpl.tmp
-	mv $(base_dir)/lorax_templates/$*.tmpl{.tmp,}
-	sed 's/@IMAGE_TAG@/$(image_tag)/'                          $(base_dir)/lorax_templates/$*.tmpl > $(base_dir)/lorax_templates/$*.tmpl.tmp
-	mv $(base_dir)/lorax_templates/$*.tmpl{.tmp,}
-	sed 's/@IMAGE_REPO_ESCAPED@/$(image_repo_double_escaped)/' $(base_dir)/lorax_templates/$*.tmpl > $(base_dir)/lorax_templates/$*.tmpl.tmp
-	mv $(base_dir)/lorax_templates/$*.tmpl{.tmp,}
+	sed 's/@IMAGE_NAME@/$(IMAGE_NAME)/'                        $(_BASE_DIR)/lorax_templates/$*.tmpl.in > $(_BASE_DIR)/lorax_templates/$*.tmpl
+
+	sed 's/@IMAGE_TAG@/$(IMAGE_TAG)/'                          $(_BASE_DIR)/lorax_templates/$*.tmpl > $(_BASE_DIR)/lorax_templates/$*.tmpl.tmp
+	mv $(_BASE_DIR)/lorax_templates/$*.tmpl{.tmp,}
+	
+	sed 's/@IMAGE_REPO_ESCAPED@/$(_IMAGE_REPO_DOUBLE_ESCAPED)/' $(_BASE_DIR)/lorax_templates/$*.tmpl > $(_BASE_DIR)/lorax_templates/$*.tmpl.tmp
+	mv $(_BASE_DIR)/lorax_templates/$*.tmpl{.tmp,}
 
 
 
+# Step 5: Generate xorriso input
 xorriso/input.txt: xorriso/gen_input.sh
-	bash $(base_dir)/xorriso/gen_input.sh | tee $(base_dir)/xorriso/input.txt
+	bash $(_BASE_DIR)/xorriso/gen_input.sh | tee $(_BASE_DIR)/xorriso/input.txt
 
+# Step 4: Generate xorriso script
```

Comment:
Could we move this step to be before step #5? It logically flows better if the steps are in order

Reply:
I completely forgot to do that. However, now I'm wondering should we do it in order or reverse order. The first entry is the default to run when just `make` is run, so that has to be resulting ISO. We can either start with the targets we expect users to run (the resulting ISOs) and then list the steps in order or we can start with the resulting target and then list the steps in reverse order so the entire file reads bottom up. Which way do you think would be better?